sky/task.py

```diff
@@ -579,6 +617,12 @@ def set_file_mounts(self, file_mounts: Optional[Dict[str, str]]) -> None:
                             f'File mount source {source!r} does not exist '
                             'locally. To fix: check if it exists, and correct '
                             'the path.')
+            if (target == constants.SKY_REMOTE_WORKDIR and
+                    self.workdir is not None):
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'Cannot mount to {constants.SKY_REMOTE_WORKDIR!r} '
```

Comment:
nit: Cannot use {constants.SKY_REMOTE_WORKDIR!r} as a destination path of a file mount, as it will be used by the workdir.

Actually, why don't we always allow it (so if workdir is specified, users ensure conflict-freeness; this may be useful for uploading some local files outside of local workdir to the remote workdir) -- the status quo --, or always ban it?  I find the conditional ban a bit strange, because a second `sky launch` on an alive cluster may conditionally trigger this.